src/lib.rs

```diff
@@ -201,6 +204,23 @@ impl DeviceHandle {
         self.device_type
     }
 
+    /// The [`HidDevice`] for the device.
+    #[must_use]
+    pub fn hid_device(&self) -> &HidDevice {
+        &self.hid_device
+    }
+
+    /// Returns the serial number of the device.
+    pub fn serial_number(&self) -> DeviceResult<&str> {
+        match self.hid_device.get_device_info() {
+            Ok(device_info) => match device_info.serial_number() {
+                Some(serial_number) => Ok(serial_number),
+                None => Err(DeviceError::NoSerial),
```

Comment:
I'm not a Rust expert: do you think this is the most idiomatic way to handle a missing serial number? Should we change the return type to `Option<&str>` instead? Interesting to hear your thoughts!

Reply:
It looks like that is what the `hidapi` crate does, so I wonder if we should just stick with the same: https://docs.rs/hidapi/latest/hidapi/struct.DeviceInfo.html#method.serial_number

I guess another option would be to make this an officially unexpected case, and panic?

Reply:
Thanks for the swift reply! I'm no rust expert either, just started learning it.
For my use-case I require a serial number, so I treated a lack of it as an error. But I have no objections of making it Optional.
Let me know how you would like to see it.

Reply:
> It looks like that is what the `hidapi` crate does, so I wonder if we should just stick with the same: https://docs.rs/hidapi/latest/hidapi/struct.DeviceInfo.html#method.serial_number
> 
> I guess another option would be to make this an officially unexpected case, and panic?

I think going the way `hidapi` does it would be better & consistent.
And in case the serial number is required (e.g. my use-case), the application should decide to panic or not.

Reply:
another thought crossed my mind:
In theory, a hardware device returned by `hidapi` may not have a serial number, that's why I guess it's optional. But can a Litra device not have a serial number?